src/test/java/com/open/boot/core/utils/CommonUtilTest.java

```diff
@@ -0,0 +1,107 @@
+
+        Assert.assertFalse(CommonUtil.isEmpty("foo"));
+    }
+
+    @Test
+    public void testGetStringSimple() {
+        Assert.assertEquals("foob***obar",
+                CommonUtil.getStringSimple("foobar"));
+    }
+
+    @Test
+    public void testGetYestoday() {
+        Date date =
+                new Date(System.currentTimeMillis() - 24 * 60 * 60 * 1000);
+        String data = new SimpleDateFormat("yyyyMMdd").format(date);
+        Assert.assertEquals(data, CommonUtil.getYestoday());
+    }
+
+    @Test
+    public void testGetCurrentDate() {
+        Date date = new Date();
+        String data = new SimpleDateFormat("yyyyMMdd").format(date);
+        Assert.assertEquals(data, CommonUtil.getCurrentDate());
+    }
+
+    @Test
+    public void testGetCurrentTime() {
+        Date date = new Date();
+        String data = new SimpleDateFormat("yyyyMMddHHmmss").format(date);
+        Assert.assertEquals(data, CommonUtil.getCurrentTime());
+    }
+
+    @Test
+    public void testGetGenderByIdNumber() {
+        Assert.assertEquals(1,
+                (int) CommonUtil.getGenderByIdNumber("123456789012345"));
+        Assert.assertEquals(1,
+                (int) CommonUtil.getGenderByIdNumber("123456789012345678"));
+    }
+
+    @Test
+    public void testGetBirthdayByIdNumber() {
+        Assert.assertEquals("1978-90-12",
+                CommonUtil.getBirthdayByIdNumber("123456789012345"));
+        Assert.assertEquals("7890-12-34",
+                CommonUtil.getBirthdayByIdNumber("123456789012345678"));
+    }
+
+    @Test
+    public void testGenRandomNum() {
+        Assert.assertEquals("", CommonUtil.genRandomNum(0));
+    }
+
+    @Test
+    public void testGetMapValue() {
+        HashMap<String, Object> hashMap = new HashMap<>();
+        hashMap.put("foo", "123");
+        Assert.assertEquals("123", CommonUtil.getMapValue(hashMap, "foo"));
+        Assert.assertEquals("", CommonUtil.getMapValue(hashMap, "baz"));
+    }
+
+    @Test
+    public void testGetSpecifiedDayBefore() {
+        Assert.assertEquals("2019-05-14",
+                CommonUtil.getSpecifiedDayBefore("2019-05-15"));
+    }
+
+    @Test
+    public void testGetSpecifiedDayAfter() {
+        Assert.assertEquals("2019-05-16",
+                CommonUtil.getSpecifiedDayAfter("2019-05-15"));
+    }
+
+    @Test
+    public void testGetTimesmorning() {
+        Date date = new Date();
+        date.setHours(0);
+        date.setMinutes(0);
+        date.setSeconds(0);
+        String data = new SimpleDateFormat("yy-MM-dd HH:mm:ss").format(date);
+        Assert.assertEquals(data, CommonUtil.getTimesmorning());
+    }
+
+    @Test
+    public void testGetTimesnight() {
+        Date date = new Date();
+        date.setHours(24);
+        date.setMinutes(0);
+        date.setSeconds(0);
+        String data = new SimpleDateFormat("yy-MM-dd HH:mm:ss").format(date);
+        Assert.assertEquals(data, CommonUtil.getTimesnight());
```

Comment:
This test and `testGetTimesmorning` are both a bit awkward because they effectively just do exactly what the tested method does again (albeit using slightly different stdlib classes). I'm not sure what the solution is here, it might just be that these methods aren't really appropriate for testing, or that the only good way to test them would be by mocking `Calendar`.

Reply:
Project uses mockito 2.15 and produces this error `org.powermock.reflect.exceptions.FieldNotFoundException: No instance field named "reporter" could be found in the class hierarchy of org.mockito.internal.MockitoCore.`

Reply:
Done